src/mandelbrot.rs

```diff
@@ -80,19 +127,21 @@ fn main() {
     for i in 0..size {
         xloc[i / VLEN].0[i % VLEN] = i as f64 * inv - 1.5;
     }
-    let xloc = Arc::new(xloc);
-    let pool = CpuPool::new_num_cpus();
+    let xloc2 = xloc.iter().map(|&x| x * x).collect::<Vec<_>>();
```

Comment:
having a vec of tuple instead of zipping xloc and xloc2 may improve locality, and thus cache usage.